### ISSUE TITLE
[Bug] Empty `top_logprobs` in LogProbs Output for Meta-Llama-3.1-8B-Instruct Model when Using OpenAI Compatible API

### ISSUE THREAD

**Issue by GuanghaoYe** [sgl-project/sglang#1176]
### Checklist

- [X] 1. I have searched related issues but cannot get the expected help.
- [X] 2. The bug has not been fixed in the latest version.
- [X] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [X] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [X] 5. Please use English, otherwise it will be closed.

## Description
While using SGLang to serve the `meta-llama/Meta-Llama-3.1-8B-Instruct` model, I occasionally encounter an issue where the `top_logprobs` returned in the LogProbs output is an empty set. This occurs even when the rest of the output appears to be normal.

##Configuration
The issue occurs under the following configuration:
```Python3
client.completions.create(
            model="default",
            prompt=batch_prompts,
            max_tokens=4096,
            temperature=1.0,
            logprobs=1,
            seed=42,
            n=4,
        )
```

## Expected Behavior:
The `top_logprobs` should return a set of size logprobs (in the case of configuration above 1) with valid log probability values for the token at the specified index

### Reproduction

```bash
python -m sglang.launch_server --model-path meta-llama/Meta-Llama-3.1-8B-Instruct --port 30000 --dp 4 --chunked-prefill-size 4096 --mem-fraction-static 0.8
```

### Environment

```
Python: 3.10.14 (main, May  6 2024, 19:42:50) [GCC 11.2.0]
CUDA available: True
GPU 0,1,2,3,4,5,6,7: NVIDIA H100 80GB HBM3
GPU 0,1,2,3,4,5,6,7 Compute Capability: 9.0
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 12.1, V12.1.105
CUDA Driver Version: 535.86.10
PyTorch: 2.4.0+cu121
sglang: 0.2.13
flashinfer: 0.1.5+cu121torch2.4
triton: 3.0.0
transformers: 4.44.0
requests: 2.32.3
tqdm: 4.66.5
numpy: 1.26.4
aiohttp: 3.10.5
fastapi: 0.112.1
hf_transfer: 0.1.8
huggingface_hub: 0.24.6
interegular: 0.3.3
packaging: 24.1
PIL: 10.4.0
psutil: 6.0.0
pydantic: 2.8.2
uvicorn: 0.30.6
uvloop: 0.20.0
zmq: 26.1.1
vllm: 0.5.4
multipart: 0.0.9
openai: 1.41.0
anthropic: 0.34.0
NVIDIA Topology: 
	[4mGPU0	GPU1	GPU2	GPU3	GPU4	GPU5	GPU6	GPU7	NIC0	NIC1	NIC2	NIC3	NIC4	NIC5	NIC6	NIC7	NIC8	CPU Affinity	NUMA Affinity	GPU NUMA ID[0m
GPU0	 X 	NV18	NV18	NV18	NV18	NV18	NV18	NV18	NODE	NODE	SYS	NODE	SYS	SYS	SYS	NODE	NODE	0-47	0		N/A
GPU1	NV18	 X 	NV18	NV18	NV18	NV18	NV18	NV18	NODE	NODE	SYS	NODE	SYS	SYS	SYS	NODE	NODE	0-47	0		N/A
GPU2	NV18	NV18	 X 	NV18	NV18	NV18	NV18	NV18	NODE	NODE	SYS	NODE	SYS	SYS	SYS	NODE	NODE	0-47	0		N/A
GPU3	NV18	NV18	NV18	 X 	NV18	NV18	NV18	NV18	NODE	NODE	SYS	NODE	SYS	SYS	SYS	NODE	NODE	0-47	0		N/A
GPU4	NV18	NV18	NV18	NV18	 X 	NV18	NV18	NV18	SYS	SYS	NODE	SYS	NODE	NODE	NODE	SYS	SYS	48-95	1		N/A
GPU5	NV18	NV18	NV18	NV18	NV18	 X 	NV18	NV18	SYS	SYS	NODE	SYS	NODE	NODE	NODE	SYS	SYS	48-95	1		N/A
GPU6	NV18	NV18	NV18	NV18	NV18	NV18	 X 	NV18	SYS	SYS	NODE	SYS	NODE	NODE	NODE	SYS	SYS	48-95	1		N/A
GPU7	NV18	NV18	NV18	NV18	NV18	NV18	NV18	 X 	SYS	SYS	NODE	SYS	NODE	NODE	NODE	SYS	SYS	48-95	1		N/A
NIC0	NODE	NODE	NODE	NODE	SYS	SYS	SYS	SYS	 X 	NODE	SYS	NODE	SYS	SYS	SYS	NODE	NODE				
NIC1	NODE	NODE	NODE	NODE	SYS	SYS	SYS	SYS	NODE	 X 	SYS	NODE	SYS	SYS	SYS	NODE	NODE				
NIC2	SYS	SYS	SYS	SYS	NODE	NODE	NODE	NODE	SYS	SYS	 X 	SYS	NODE	NODE	NODE	SYS	SYS				
NIC3	NODE	NODE	NODE	NODE	SYS	SYS	SYS	SYS	NODE	NODE	SYS	 X 	SYS	SYS	SYS	NODE	NODE				
NIC4	SYS	SYS	SYS	SYS	NODE	NODE	NODE	NODE	SYS	SYS	NODE	SYS	 X 	NODE	NODE	SYS	SYS				
NIC5	SYS	SYS	SYS	SYS	NODE	NODE	NODE	NODE	SYS	SYS	NODE	SYS	NODE	 X 	NODE	SYS	SYS				
NIC6	SYS	SYS	SYS	SYS	NODE	NODE	NODE	NODE	SYS	SYS	NODE	SYS	NODE	NODE	 X 	SYS	SYS				
NIC7	NODE	NODE	NODE	NODE	SYS	SYS	SYS	SYS	NODE	NODE	SYS	NODE	SYS	SYS	SYS	 X 	NODE				
NIC8	NODE	NODE	NODE	NODE	SYS	SYS	SYS	SYS	NODE	NODE	SYS	NODE	SYS	SYS	SYS	NODE	 X 				

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: mlx5_0
  NIC1: mlx5_1
  NIC2: mlx5_2
  NIC3: mlx5_3
  NIC4: mlx5_4
  NIC5: mlx5_5
  NIC6: mlx5_6
  NIC7: mlx5_7
  NIC8: mlx5_8


ulimit soft: 4096
```

**Comment by hnyls2002** [comment]
@GuanghaoYe 
1. Is the empty set in the input tokens or in the output tokens? With Openai API, the input token's information about logprobs is not calculated, the same as with Openai. If this is the case, you can try with SGLang's /generate API.
2. Could you provide a minimal reproducible script?

**Comment by GuanghaoYe** [comment]
@hnyls2002 Thanks for the reply. It's in the output tokens. I am currently working to find a small example for reproduce the bug.

**Comment by merrymercy** [comment]
It should be fixed in the latest version v0.3.1. Please open a new issue if it still exists in the latest version.
